src/components/Tree/Node.tsx

```diff
@@ -25,67 +27,93 @@ type NodeProps = {
 };
 
 export const TreeNode = ({
-    node: { id, value, name, label, icon, nodes },
+    node: { id, value, name, label, icon, nodes, actions },
     strong = false,
     activeNodeId = null,
     onClick,
     parentIds = [],
 }: NodeProps): ReactElement<NodeProps> => {
     const [showNodes, setShowNodes] = useState(false);
+    const [isHovered, setIsHovered] = useState(false);
     const selected = id === activeNodeId;
 
+    const setHoveredTrue = () => setIsHovered(true);
+    const setHoveredFalse = () => setIsHovered(false);
+    const onNodeClick = () => {
+        if (!value) {
+            setShowNodes(!showNodes);
+            return;
+        }
+
+        if (onClick) {
+            onClick(activeNodeId === id ? null : id);
```

Comment:
activeNodeId === id can be replaced with selected

Reply:
Done